data/modules/CargoRun/CargoRun.lua

```diff
@@ -0,0 +1,1136 @@
+local custom_cargo_weight_sum = 0
+for branch,branch_array in pairs(custom_cargo) do
+	custom_cargo[branch].weight = #branch_array.goods
+	custom_cargo_weight_sum = custom_cargo_weight_sum + #branch_array.goods
+end
+
+local ads = {}
+local missions = {}
+
+local isQualifiedFor = function(reputation, ad)
+	return
+		reputation >= 12 or
+		ad.localdelivery or
+		(ad.risk <  0.1 and ad.urgency <= 0.1) or
+		(ad.risk <  0.5 and ad.urgency <= 0.5 and reputation >= 4) or
+		false
+end
+
+-- This function returns the number of flavours of the given string str
+-- It is assumed that the first flavour has suffix '_1'
+local getNumberOfFlavours = function (str)
+	local num = 1
+
+	while l[str .. "_" .. num] do
+		num = num + 1
+	end
+	return num - 1
+end
+
+local onChat = function (form, ref, option)
+	local ad = ads[ref]
+
+	form:Clear()
+
+	if option == -1 then
+		form:Close()
+		return
+	end
+
+	local qualified = isQualifiedFor(Character.persistent.player.reputation, ad)
+
+	form:SetFace(ad.client)
+
+	if not qualified then
+		form:SetMessage(l["DENY_" .. Engine.rand:Integer(1, getNumberOfFlavours("DENY"))])
+		return
+	end
+
+	if option == 0 then
+		local introtext = string.interp(ad.introtext, {
+			name		= ad.client.name,
+			cash		= Format.Money(ad.reward),
+			cargoname	= ad.cargotype:GetName(),
+			starport	= ad.location:GetSystemBody().name,
+			system		= ad.location:GetStarSystem().name,
+			sectorx		= ad.location.sectorX,
+			sectory		= ad.location.sectorY,
+			sectorz		= ad.location.sectorZ,
+			dom_starport	= ad.domicile:GetSystemBody().name,
+			dom_system	= ad.domicile:GetStarSystem().name,
+			dom_sectorx	= ad.domicile.sectorX,
+			dom_sectory	= ad.domicile.sectorY,
+			dom_sectorz	= ad.domicile.sectorZ,
+			dist		= string.format("%.2f", ad.dist),
+		})
+		form:SetMessage(introtext)
+
+	elseif option == 1 then
+		local n = getNumberOfFlavours("WHYSOMUCH_" .. ad.branch)
+		if n >= 1 then
+			form:SetMessage(string.interp(l["WHYSOMUCH_" .. ad.branch .. "_" .. Engine.rand:Integer(1, n)], { cargoname = ad.cargotype:GetName() }))
+		elseif ad.urgency >= 0.8 then
+			form:SetMessage(string.interp(l["WHYSOMUCH_URGENT_" .. Engine.rand:Integer( 1, getNumberOfFlavours("WHYSOMUCH_URGENT"))], { cargoname = ad.cargotype:GetName() }))
+		else
+			form:SetMessage(string.interp(l["WHYSOMUCH_" .. Engine.rand:Integer( 1, getNumberOfFlavours("WHYSOMUCH"))], { cargoname = ad.cargotype:GetName() }))
+		end
+
+	elseif option == 2 then
+		local howmuch
+		if ad.wholesaler then
+			howmuch = string.interp(l["HOWMUCH_WHOLESALER_" .. Engine.rand:Integer(1, getNumberOfFlavours("HOWMUCH_WHOLESALER"))], {
+				amount    = ad.amount,
+				cargoname = ad.cargotype:GetName(),
+			})
+		else
+			local num = ad.amount <= 3 and getNumberOfFlavours("HOWMUCH") or (getNumberOfFlavours("HOWMUCH") - 2) -- The last 2 strings contain "only"
+			howmuch = string.interp(l["HOWMUCH_" .. Engine.rand:Integer(1, num)], {
```

Comment:
I think the comparison should be the other way around, `ad.amount >= 3`. I've rewritten it now, so don't worry about it, I've made some changes I merge into your commits, and then I merge this into master shortly, so I've got this from now.

Also, I've changed this to be `HOWMUCH_SINGULAR_` and `HOWMUCH_` since I think the distinction between is/are is more importnat than adding an "only" if it's below 3t. So now if it's just 1t it says "is only..." or uses the plural strings.